### ISSUE TITLE
Incompatible with v0.15b0 of opentelemetry-api, -sdk

### ISSUE THREAD

**Issue by spencerwilson-optimizely** [honeycombio/opentelemetry-exporter-python#10]
opentelemetry-ext-honeycomb v0.5b0 claims compatibility with `>=0.5b0` versions of opentelemetry-api and opentelemetry-sdk ([source](https://github.com/honeycombio/opentelemetry-exporter-python/blob/v0.5b0/setup.cfg#L28-L29)), but it's actually incompatible with version v0.15b0 of those packages.

Not sure but this may be the cause of #9.

I suppose the periodic CI runs wouldn't have caught this regression if they're respecting the poetry.lock (which is [pinning](https://github.com/honeycombio/opentelemetry-exporter-python/blob/98e593f43c6e5903a97d6442cc4bc817e9319381/poetry.lock#L134-L154) these to 0.14b0). Pros and cons of reproducible builds, I guess 🤷‍♂️ 

## Repro steps
In requirements.txt:
```
opentelemetry-api == v0.15b0
opentelemetry-ext-honeycomb == 0.5b0
opentelemetry-sdk == v0.15b0
```
then run
```py
from opentelemetry.ext.honeycomb import HoneycombSpanExporter
```
and observe
```sh
Traceback (most recent call last):
  File "main.py", line 4, in <module>
    from opentelemetry.ext.honeycomb import HoneycombSpanExporter
  File "/Users/swilson/.pyenv/versions/3.8.6/lib/python3.8/site-packages/opentelemetry/ext/honeycomb/__init__.py", line 30, in <module>
    from opentelemetry.trace.status import StatusCanonicalCode
ImportError: cannot import name 'StatusCanonicalCode' from 'opentelemetry.trace.status' (/Users/swilson/.pyenv/versions/3.8.6/lib/python3.8/site-packages/opentelemetry/trace/status.py)
```
Changing the opentelemetry package versions to v0.14b0 resolves the issue; `HoneycombSpanExporter` is importable.

## Suggested fix
One of

- add `< 0.15b0` constraints tro pyproject.toml, to explicitly declare incompatibility;
- update the code to only support, and the constraints to say, `>= 0.15b0`; or
- update the code to have dual compatibility

**Comment by lizthegrey** [comment]
oh, whoops. I probably should have pinned v0.5b0 to v0.5b0 etc. to keep things in sync rather than offer open-ended compat. it worked up through v0.14, as a consolation!

Putting this in the queue for the integrations team here, as we now have folks dedicated to working on OTel :)

**Comment by paulosman** [comment]
Thanks for bringing this to our attention, @spencerwilson-optimizely ! If you could give it another shot, that'd be appreciated - opted to require `0.15b0` or above now.

**Comment by lizthegrey** [comment]
we've just released v0.15b0, please let us know if your issue is solved.

**Comment by spencerwilson-optimizely** [comment]
Confirmed that the issue's no longer reproducible with v0.15b1! (0.15b0 isn't on PyPI, prob for reasons of the naming issue lizf fixed). This combo works:
```
opentelemetry-api == v0.15b0
opentelemetry-ext-honeycomb == 0.15b1
opentelemetry-sdk == v0.15b0
```

Thanks, @paulosman and @lizthegrey !
